libinfinity/common/inf-file-util.c

```diff
@@ -517,4 +517,49 @@ inf_file_util_delete(const gchar* path,
     return inf_file_util_delete_file(path, error);
 }
 
+/**
+ * inf_file_write_data_with_perms:
+ * @filename: Filename of the file to be written to.
+ * @data: Data that should be written to file.
+ * @length: Length of data in bytes.
+ * @flags: File access mode flags of the file to be open()'ed.
+ * @mode: File permission bits of the file to be open()'ed.
+ * @error: Location to store error information, if any, or %NULL.
+ *
+ * Writes @data to the file pointed to by @filename. Files may be created with
+ * specified permissions (@mode). If the function fails, %FALSE is returned
+ * and @error is set.
+ *
+ * Returns: %TRUE on success or %FALSE on error.
+ */
+gboolean
+inf_file_write_data_with_perms(const gchar* filename,
+                               const void* data,
+                               size_t length,
+                               int flags,
```

Comment:
I wonder if flags should really be an argument to this function given that it attempts to write to the newly created file immediately afterwards, and so passing or not passing certain flags does not make sense for it (such as everything that would result in the file being opened for reading only). In both cases this function is used, the flags are the same anyway, so we could just hardcode those inside this function.